src/Agent.Listener/JobDispatcher.cs

```diff
@@ -632,17 +641,33 @@ private async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockTok
                     }
                     else
                     {
-                        // retry till reach lockeduntil
-                        remainingTime = request.LockedUntil.Value - DateTime.UtcNow;
+                        // retry till reach lockeduntil + 5 mins oauth token time skew limit + 5 mins extra buffer for the delay of request monitor job on server side.
+                        remainingTime = request.LockedUntil.Value + TimeSpan.FromMinutes(5) + TimeSpan.FromMinutes(5) - DateTime.UtcNow;
```

Comment:
it looks like `remainingTime` should just be a Boolean instead of a TimeSpan. Perhaps `hasRemainingTime` would be a better name for a Boolean.